terraform/dotnet/ec2/asg/main.tf

```diff
@@ -0,0 +1,263 @@
+  instance_type               = "t3.micro"
+  key_name                    = local.ssh_key_name
+  associate_public_ip_address = true
+  iam_instance_profile        = "APP_SIGNALS_EC2_TEST_ROLE"
+  security_groups             = [aws_default_vpc.default.default_security_group_id]
+
+  user_data = <<-EOF
+    #!/bin/bash
+    set -o errexit
+    cd /home/ec2-user
+
+    # Install DotNet and wget
+    sudo yum install -y wget
+    sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+    sudo wget -O /etc/yum.repos.d/microsoft-prod.repo https://packages.microsoft.com/config/fedora/37/prod.repo
+    sudo dnf install -y dotnet-sdk-8.0
+    sudo yum install unzip -y
+
+    # Copy in CW Agent configuration
+    agent_config='${replace(replace(file("./amazon-cloudwatch-agent.json"), "/\\s+/", ""), "$REGION", var.aws_region)}'
+    echo $agent_config > amazon-cloudwatch-agent.json
+
+    # Get and run CW agent rpm
+    ${var.get_cw_agent_rpm_command}
+    sudo rpm -U ./cw-agent.rpm
+    sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:./amazon-cloudwatch-agent.json
+
+    # Get ADOT distro and unzip it
+    ${var.get_adot_distro_command}
+
+    # Get and run the sample application with configuration
+    aws s3 cp ${var.sample_app_zip} ./dotnet-sample-app.zip
+    unzip -o dotnet-sample-app.zip
+
+    # Get Absolute Path
+    current_dir=$(pwd)
+
+    # Export environment variables for instrumentation
+    cd ./asp_frontend_service
+    export INSTALL_DIR=../dotnet-distro
+    export CORECLR_ENABLE_PROFILING=1
+    export CORECLR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
+    export CORECLR_PROFILER_PATH=$current_dir/dotnet-distro/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
+    export DOTNET_ADDITIONAL_DEPS=$current_dir/dotnet-distro/AdditionalDeps
+    export DOTNET_SHARED_STORE=$current_dir/dotnet-distro/store
+    export DOTNET_STARTUP_HOOKS=$current_dir/dotnet-distro/net/OpenTelemetry.AutoInstrumentation.StartupHook.dll
+    export OTEL_DOTNET_AUTO_HOME=$current_dir/dotnet-distro
+    export OTEL_DOTNET_AUTO_PLUGINS="AWS.Distro.OpenTelemetry.AutoInstrumentation.Plugin, AWS.Distro.OpenTelemetry.AutoInstrumentation"
+    export OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf
+    export OTEL_EXPORTER_OTLP_ENDPOINT=http://127.0.0.1:4316
+    export OTEL_AWS_APPLICATION_SIGNALS_EXPORTER_ENDPOINT=http://127.0.0.1:4316/v1/metrics
+    export OTEL_METRICS_EXPORTER=none
+    export OTEL_RESOURCE_ATTRIBUTES=service.name=dotnet-sample-application-${var.test_id}
+    export OTEL_AWS_APPLICATION_SIGNALS_ENABLED=true
+    export OTEL_TRACES_SAMPLER=always_on
+    export OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://127.0.0.1:4316/v1/traces
+    export OTEL_LOG_LEVEL=debug
+    export OTEL_DOTNET_AUTO_LOG_DIRECTORY=/tmp
+    export ASPNETCORE_URLS=http://0.0.0.0:8080
+    sudo -E dotnet build
+    sudo -E nohup dotnet bin/Debug/netcoreapp8.0/asp_frontend_service.dll &
```

Comment:
Have you tried using log integration? It would be a good improvement for future debugging if needed

Reply:
I believe there is no log (or only minor log) for asp_frontend and backend?